feat(hogql): Support modifying csv allow double quotes settings

tomasfarias opened the pull request:
## Problem

The format_csv_allow_double_quotes ClickHouse setting is defaulted to
0. But, as CSV is a very shaky format, this can cause lots of trouble
when CSVs are not formatted in certain ways, particularly when JSON
values are nested in CSVs.

## Changes

We should ultimately allow users to customize how do their CSV files
look like, but it seemed faster for now to allow configuring this on a
team by team basis.

So, I've added a new HogQLModifier that when set will update the
aforementioned setting.

The plan is to then set this modifier for teams who request it.

Looking for more seasoned HogQL developers to tell me if this is the
right approach! My thought was that eventually this modifier could be
somewhat set at the `ExternalDataSource` level, and the user can be 
prompted to customize their CSV experience when syncing one. But that
seemed quite involved and I wasn't sure how to achieve that. So, I've 
left it as an enhancement for later.

tomasfarias committed:
feat(hogql): Support modifying csv allow double quotes settings

The format_csv_allow_double_quotes ClickHouse setting is defaulted to
0. But, as CSV is a very shaky format, this can cause lots of trouble
when CSVs are not formatted in certain ways, particularly when JSON
values are nested in CSVs.

We should ultimately allow users to customize how do their CSV files
look like, but it seemed faster for now to allow configuring this on a
team by team basis.

So, I've added a new HogQLModifier that when set will update the
aforementioned setting.

The plan is to then set this modifier for teams who request it.

tomasfarias committed:
chore: Add new modifier to magic schema files

tomasfarias committed:
chore: Sort modifiers

EDsCODE reviewed (commented):
How will this be set per team?

tomasfarias commented:
> How will this be set per team?

By manually modifying the team object.

tomasfarias commented:
The idea is that eventually this could be set at the source or table level. The manual team object modification is just a quick fix.

tomasfarias committed:
chore: Update modifier name to better match setting

The setting name starts with "format_" so I'm making the modifier
match that more closely for consistency.
